SharedUtilities.js

```diff
@@ -279,51 +287,44 @@ function getValidSubmissionEmails(submissionSheet) {
 
   const { submissionWindowStart, submissionWindowEnd } = getSubmissionWindowTimes();
 
-  // Dynamically get column indices by header name
-  // TODO Confirm: FYI - changed to Submission form, not generic google form email
-  const submitterEmailColumnIndex = getColumnIndexByName(submissionSheet, SUBM_FORM_USER_PROVIDED_EMAIL_COLUMN);
-  const submitterTimestampColumnIndex = getColumnIndexByName(submissionSheet, GOOGLE_FORM_TIMESTAMP_COLUMN);
+  const headers = submissionSheet.getRange(1, 1, 1, submissionSheet.getLastColumn()).getValues()[0];
+  const submitterEmailColumnIndex = headers.indexOf(SUBM_FORM_USER_PROVIDED_EMAIL_COLUMN) + 1;
+  const submitterTimestampColumnIndex = headers.indexOf(GOOGLE_FORM_TIMESTAMP_COLUMN) + 1;
 
-  if (submitterEmailColumnIndex === -1 || submitterTimestampColumnIndex === -1) {
-    Logger.log(
-      `Error: Required columns (Email: ${SUBM_FORM_USER_PROVIDED_EMAIL_COLUMN}, Timestamp: ${GOOGLE_FORM_TIMESTAMP_COLUMN}) not found.`
-    );
+  if (submitterEmailColumnIndex === 0 || submitterTimestampColumnIndex === 0) {
+    Logger.log(`Error: Required columns not found.`);
     return [];
   }
 
-  // Extract valid requests within the submission time window
   const validSubmitters = submissionSheet
     .getRange(2, 1, lastRow - 1, submissionSheet.getLastColumn())
     .getValues()
     .filter((row, index) => {
       const submissionTimestamp = new Date(row[submitterTimestampColumnIndex - 1]);
-      const submitterEmail = row[submitterEmailColumnIndex - 1]?.trim();
+      const submitterEmail = row[submitterEmailColumnIndex - 1]?.trim().toLowerCase();
 
-      // Validate timestamp and email
-      if (!submissionTimestamp || isNaN(submissionTimestamp)) {
-        Logger.log(`Row ${index + 2}: Invalid timestamp.`);
+      if (!submissionTimestamp || !submitterEmail) {
+        Logger.log(`Row ${index + 2}: Missing timestamp or email.`);
         return false;
       }
 
-      if (!submitterEmail) {
-        Logger.log(`Row ${index + 2}: Missing email.`);
+      const isWithinWindow = submissionTimestamp >= submissionWindowStart && submissionTimestamp <= submissionWindowEnd;
+
+      if (!isWithinWindow) {
```

Comment:
Filter these above so we don't have to loop through all submissions every single month for the rest of all time. This function is already slow, it will get slower and slower every month. We don't need to log each one that doesn't meet the filter, just ignore them (filter earlier.)

Reply:
withdrawing/resolving comment for now.